aspnetcore/migration/webapi.md

```diff
@@ -4,47 +4,169 @@ author: ardalis
+
+## Migrate configuration
+
+ASP.NET Core doesn't use the *App_Start* folder or the *Global.asax* file, and the *web.config* file is added at publish time. *Startup.cs* is the replacement for *Global.asax* and is located in the project root. The `Startup` class handles all app startup tasks. For more information, see <xref:fundamentals/startup>.
+
+All ASP.NET Core templates include attribute routing in the generated code. The following `UseRouting` and `UseEndpoints` calls register route matching and endpoint execution in the [middleware](xref:fundamentals/middleware/index) pipeline.  They replace the *ProductsApp* project's *App_Start/WebApiConfig.cs* file:
+
+[!code-csharp[](webapi/sample/3.x/ProductsCore/Startup.cs?name=snippet_Configure&highlight=10,14])]
+
+## Migrate models and controllers
+
+Copy over the *ProductApp* project's controller and the model it uses. Follow these steps:
+
+1. Copy *Controllers/ProductsController.cs* from the original project to the new one.
+1. Copy the entire *Models* folder from the original project to the new one.
+1. Change the copied files' namespaces to match the new project name (*ProductsCore*). Adjust the `using ProductsApp.Models;` statement in *ProductsController.cs* too.
+
+At this point, building the app results in a number of compilation errors. The errors occur because the following components don't exist in ASP.NET Core:
+
+* `ApiController` class
+* `System.Web.Http` namespace
+* `IHttpActionResult` interface
+
+Fix the errors as follows:
+
+1. Change `ApiController` to <xref:Microsoft.AspNetCore.Mvc.ControllerBase>. Add `using Microsoft.AspNetCore.Mvc;` to resolve the `ControllerBase` reference.
+1. Delete `using System.Web.Http;`.
+1. Change the `GetProduct` action's return type from `IHttpActionResult` to `ActionResult<Product>`.
+
+Simplify the `GetProduct` action's `return` statement to the following:
+
+```csharp
+return product;
+```
+
+## Configure routing
+
+Configure routing as follows:
+
+1. Mark the `ProductsController` class with the following attributes:
+
+    ```csharp
+    [Route("api/[controller]")]
+    [ApiController]
+    ```
+
+    The preceding [`[Route]`](xref:Microsoft.AspNetCore.Mvc.RouteAttribute) attribute configures the controller's attribute routing pattern. The [`[ApiController]`](xref:Microsoft.AspNetCore.Mvc.ApiControllerAttribute) attribute makes attribute routing a requirement for all actions in this controller.
+
+    Attribute routing supports tokens, such as `[controller]` and `[action]`. At runtime, each token is replaced with the name of the controller or action, respectively, to which the attribute has been applied. The tokens reduce the number of magic strings in the project. The tokens also ensure routes remain synchronized with the corresponding controllers and actions when automatic rename refactorings are applied.
+1. Set the project's compatibility mode to ASP.NET Core 3.0:
```

Comment:
This step should be removed. It has no affect in 3.0+ apps. Be sure to remove the corresponding code from the sample app too.